test/test_unit_check_logic.py

```diff
@@ -0,0 +1,57 @@
+import unittest
+from unittest.mock import MagicMock, patch
+
+from conjur_api import Client
+from conjur.logic.check_logic import CheckLogic
+from conjur_api.errors.errors import HttpStatusError
+
+class CheckLogicTest(unittest.TestCase):
+
+    client = Client
+    check_logic = CheckLogic(client)
+
+    def test_check_logic_constructor(self):
+        mock_client = None
+        mock_check_logic = CheckLogic(mock_client)
+        self.assertEquals(mock_check_logic.client, mock_client)
+
+    @patch('conjur_api.Client.check_privilege')
+    def test_check_privilege_calls_client(self, mock_check):
+        self.check_logic.check('some_kind', 'some_resource_id', 'some_privilege', 'some_role_id')
+        mock_check.assert_called_once_with('some_kind', 'some_resource_id', 'some_privilege', 'some_role_id')
+
+    def test_check_privilege_missing_resource_parameter_raises_exception(self):
+        with self.assertRaises(TypeError):
+            self.check_logic.check('some_kind', 'some_privilege')
+
+    def test_check_privilege_missing_privilege_parameter_raises_exception(self):
+        with self.assertRaises(TypeError):
+            self.check_logic.check('some_kind', 'some_resource_id')
+
+    def test_check_privilege_response_default_role(self):
+        self.client.check_privilege = MagicMock(return_value=True)
+        self.assertEqual(self.check_logic.check('some_kind', 'some_resource_id', 'some_privilege'), True)
+
+    def test_check_privilege_response_specified_role(self):
+        self.client.check_privilege = MagicMock(return_value=True)
+        self.assertEqual(self.check_logic.check('some_kind', 'some_resource_id', 'some_privilege', 'some_role_id'), True)
+
+    def test_check_privilege_response_invalid_credentials_default_role(self):
+        self.client.check_privilege = MagicMock(side_effect=HttpStatusError(status=401, message="Invalid Credentials"))
+        with self.assertRaises(HttpStatusError):
+            self.check_logic.check('some_kind', 'some_resource_id', 'some_privilege')
+
+    def test_check_privilege_response_invalid_credentials_specified_role(self):
+        self.client.check_privilege = MagicMock(side_effect=HttpStatusError(status=401, message="Invalid Credentials"))
+        with self.assertRaises(HttpStatusError):
+            self.check_logic.check('some_kind', 'some_resource_id', 'some_privilege', 'some_role_id')
+
+    def test_role_exists_raises_exception_other_http_error_default_role(self):
```

Comment:
```suggestion
    def test_check_privilege_raises_exception_other_http_error_default_role(self):
```

Reply:
On second thought we may not need this test (or the one below it). You already have two tests above where you are mocking a 401 HttpStatusError. We just care that an HttpStatusError from the API call results in an exception being raised in the CLI, and we can assume that differences in status code/message don't matter.

Reply:
removed the status=500 tests.